src/error.rs

```diff
@@ -7,6 +7,9 @@ use thiserror::Error;
 
 #[derive(Error, Debug, PartialEq)]
 pub enum MinaMeshError {
+  #[error("Invalid transaction status: '{0}'. Valid are 'applied' and 'failed'")]
+  InvalidTransactionStatus(String),
```

Comment:
@joaosreis is it okay for us to introduce new error variants that weren't in the initial Rosetta implementation?

Reply:
Just to note in Rosetta, when using invalid `status` in `/search/transactions` it throws SQL failure, which doesn't seem nice:
```
{
  "code": 1,
  "message": "SQL failure",
  "description": "We encountered a SQL failure.",
  "retriable": false,
  "details": {
    "body": [
      "Sql",
      "Request to <postgres://mina:_@archive-postgresql:5432/archive> failed: ERROR:  invalid input value for enum transaction_status:
```